### DISCUSSION TITLE
Changing width of components

### DISCUSSION THREAD

**Discussion by ChangPaul** [zauberzeug/nicegui#254] [Q&A] [answered]
Is it possible to change the width of components? In particular, can the "text input" component be wider than the default (or even automatically scaled to the full width of its container)?
Is changing the width other components, like "sliders", possible?

**Comment by falkoschindler** [comment, marked as answer]
Sure. You can either use `.style` to define [CSS](https://www.w3schools.com/css/) for an element or set `.classes` to use [Tailwind](https://tailwindcss.com/):

```python
ui.input('Name') # for reference
ui.input('Street').style('width: 200px')
ui.input('City').classes('w-20')
```

This works for most UI elements, also for sliders:

```python
ui.slider(min=0, max=100, step=1, value=50).style('width: 200px')
ui.slider(min=0, max=100, step=1, value=50).classes('w-20')
```

Using the full width of the parent element is possible with CSS or Tailwind:

```python
ui.input('Street').style('width: 100%')
ui.input('City').classes('w-full')
```
